### PR TITLE
Update Servers' Obsolete messages and comments

### PR DESCRIPTION
**Pull request by halter73**
Partially adddresses #27529

### CODE REVIEW

File: src/Servers/IIS/IIS/src/Core/IISServerAuthenticationHandler.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ namespace Microsoft.AspNetCore.Server.IIS.Core
     /// <summary>
     /// The default authentication handler with IIS In-Process
     /// </summary>
-    [Obsolete("The IISServerAuthenticationHandler is obsolete and will be removed in a future release.")]
+    [Obsolete("The IISServerAuthenticationHandler is obsolete and may be removed in a future release.")] // Never remove.
```
Comment (halter73) [inline src/Servers/IIS/IIS/src/Core/IISServerAuthenticationHandler.cs:15]: @jkotalik. This is one of the ones that I'm most unsure about. Do you see this being painful to maintain and warranting removal after .NET 6? Personally, I don't see much harm in leaving an obsolete type in the `Microsoft.AspNetCore.Server.IIS.Core` namespace, and it's nice not to break people.

File: src/Servers/IIS/IIS/src/Core/WrappingStream.cs [hunk 2]
```diff
@@ -107,7 +107,6 @@ public override int EndRead(IAsyncResult asyncResult)
         public override void EndWrite(IAsyncResult asyncResult)
             => _inner.EndWrite(asyncResult);
 
-        [Obsolete("This Remoting API is not supported and throws PlatformNotSupportedException.", DiagnosticId = "SYSLIB0010", UrlFormat = "https://aka.ms/dotnet-warnings/{0}")]
```
Comment (halter73) [inline src/Servers/IIS/IIS/src/Core/WrappingStream.cs:110]: This, and the other Obsolete attribute I removed for InitializeLifetimeService() were on internal types.

File: src/Servers/IIS/IIS/src/Core/IISServerAuthenticationHandler.cs (hunk 1 above)
Comment (jkotalik) [reply src/Servers/IIS/IIS/src/Core/IISServerAuthenticationHandler.cs:15]: I'd argue that these types were never intended to be public and shouldn't be used. I don't see why we should never remove these? 

FYI we made these obsolete in .NET 6.

Comment (halter73) [reply src/Servers/IIS/IIS/src/Core/IISServerAuthenticationHandler.cs]: I am not sure about IISServerAuthenticationHandler, but I don't see why we'd ever remove something like WriteOnlyStream which costs practically nothing to maintain, lives in [a namespace whose only public types are obsolete](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.iis.core?view=aspnetcore-5.0), and whose removal could potentially break old libraries that depend those types.

Comment (jkotalik) [reply src/Servers/IIS/IIS/src/Core/IISServerAuthenticationHandler.cs]: We already scanned for any libraries that depend on these types, and the number of hits were in the single digits across all of github and nuget.

This isn't a hill I'm willing to die on, I just think the harm of deleting these types is very low and would clean up an entire namespace that shouldn't have been public.

**Comment by halter73** [comment]
I've updated the comments on the obsolete Microsoft.AspNetCore.Server.IIS.Core types to indicate they should be removed after .NET 6.

**Comment by halter73** [comment]
@jkotalik @Tratcher ping

**Review by Tratcher — APPROVED** [review]

**Pull request merged**